tests/ext-activitylog.test.js

```diff
@@ -27,28 +27,715 @@ function observeChange(element) {
+        '.toggle-btn'
+      );
+
+      const tableBody = activityLog.view.logView.shadowRoot.querySelector(
+        'tbody'
+      );
+      const getTableRows = () => tableBody.querySelectorAll('tr');
+      let tableRows = getTableRows();
+
+      // Initially filter checkbox list is hidden
+      expect(
+        activityLog.view.extFilter.classList.contains('expanded')
+      ).toBeFalsy();
+
+      const viewTypeFilterBtnChanged = observeChange(
+        activityLog.view.viewTypeFilter
+      );
+      viewTypesFilterBtn.click();
+      await viewTypeFilterBtnChanged;
+
+      // the filter checkbox list dropdown is displayed
+      expect(
+        activityLog.view.viewTypeFilter.classList.contains('expanded')
+      ).toBeTruthy();
+
+      const otherLabledCheckbox = activityLog.view.viewTypeFilter.shadowRoot.querySelector(
+        '.filter-checkbox input[value="other"]'
+      );
+
+      expect(otherLabledCheckbox.checked).toBeTruthy();
+
+      expect(tableRows[0].hidden).toBeFalsy();
+      expect(tableRows[1].hidden).toBeFalsy();
+      expect(tableRows[2].hidden).toBeFalsy();
+
+      const tableRowPropChanged = observeChange(tableRows[2]);
+      otherLabledCheckbox.click();
+      await tableRowPropChanged;
+
+      expect(tableRows[0].hidden).toBeFalsy();
+      expect(tableRows[1].hidden).toBeFalsy();
+      // 3rd row contains the log with undefined viewType
+      expect(tableRows[2].hidden).toBeTruthy();
+    });
+  });
+
+  test('searching by keyword should search in the data object of the log to filter out the logs', async () => {
+    expect(window.customElements.get('filter-keyword')).toBe(FilterKeyword);
+
+    const logs = [
+      {
+        /* renders the 1st row in the table */
+        id: 'id1@test',
+        viewType: 'viewType1@test',
+        type: 'type1@test',
+        data: { test: 'test1@data' },
+        timeStamp: 1597686226302,
+      },
+      {
+        /* renders the 2nd row in the table */
+        id: 'id2@test',
+        viewType: 'viewType2@test',
+        type: 'type2@test',
+        data: { test: 'test2@data' },
+        timeStamp: 1597686226302,
+      },
+    ];
+
+    const addListener = jest.fn();
+    const removeListener = jest.fn();
+    const sendMessage = jest.fn();
+    const connect = jest.fn();
+
+    window.browser = {
+      runtime: {
+        onMessage: { addListener },
+        sendMessage,
+        connect,
+      },
+      menus: {
+        onClicked: { addListener, removeListener },
+        onHidden: { addListener, removeListener },
+      },
+    };
+
+    sendMessage.mockImplementation(() => {
+      return Promise.resolve({ existingLogs: [] });
+    });
+
+    document.body.innerHTML = activityLogBody;
+
+    const { activityLog } = new ActivityLog();
+
+    activityLog.handleNewLogs(logs);
+
+    const tableBody = activityLog.view.logView.shadowRoot.querySelector(
+      'tbody'
+    );
+    const getTableRows = () => tableBody.querySelectorAll('tr');
```

Comment:
@rpl This lines i.e.
```
const tableBody = activityLog.view.logView.shadowRoot.querySelector('tbody');
const getTableRows = () => tableBody.querySelectorAll('tr');
```
 is redundant in in many test cases including every tests under `filtering logs from the logs table` section. Maybe it's better to extract it as test helper.